schemas/ispyb/stored_programs/sp_retrieve_robot_actions_for_sample.sql

```diff
@@ -0,0 +1,125 @@
+DELIMITER ;;
+
+CREATE OR REPLACE DEFINER `ispyb_root`@`%` PROCEDURE `retrieve_robot_actions_for_sample`(
+	p_id int unsigned,
+	p_useRobotActionSession boolean,
+	p_authLogin varchar(45)
+)
+READS SQL DATA
+COMMENT 'Returns a multi-row result-set with the robot actions for the given sample ID.'
+BEGIN
+
+	IF p_id IS NOT NULL THEN
+
+		IF p_useRobotActionSession = True THEN
+
+			IF p_authlogin IS NOT NULL THEN
+			-- Authorise if the persion (p_authLogin) is a member of a session on the proposal.
+
+				SELECT DISTINCT 
+				ra.robotActionId "robotActionId",
+				ra.blsessionId "sessionId",
+				ra.blSampleId "sampleId",
+				ra.actionType "actionType",
+				ra.startTimeStamp "startTimeStamp",
+				ra.endTimeStamp "endTimeStamp",
+				ra.status "status",
+				ra.message "message",
+				ra.containerLocation "location",
+				ra.dewarLocation "dewarLocation",
+				ra.sampleBarcode "sampleBarcode",
+				xtalSnapshotBefore "xtalSnapshotBefore",
+				xtalSnapshotAfter "xtalSnapshotAfter"
```

Comment:
This works as is without specifying the table alias (`ra.`), but if we in the future add a column called `xtalSnapshotBefore`/`After` to one of the joined tables, then it will fail (ambiguous column), so it's best to specify the table alias just to be sure.